src/GoogleAuthenticator.php

```diff
@@ -41,47 +41,53 @@ final class GoogleAuthenticator implements GoogleAuthenticatorInterface
     /**
      * @var int
      */
-    private $codePeriod = 30;
+    private $codePeriod;
+
+    /**
+     * @var int
+     */
+    private $bufferPeriod = 30;
 
     /**
      * @param int                     $passCodeLength
      * @param int                     $secretLength
+     * @param int                     $codePeriod
      * @param \DateTimeInterface|null $now
      */
-    public function __construct(int $passCodeLength = 6, int $secretLength = 10, \DateTimeInterface $now = null)
+    public function __construct(int $passCodeLength = 6, int $secretLength = 10, int $codePeriod = 30, \DateTimeInterface $now = null)
     {
         $this->passCodeLength = $passCodeLength;
         $this->secretLength = $secretLength;
+        $this->codePeriod = $codePeriod;
+        $this->bufferPeriod = $codePeriod < $this->bufferPeriod ? $codePeriod : $this->bufferPeriod;
```

Comment:
Why do we need the bufferPeriod? Isn’t it enough to make codePeriod configurable via the constructor and use this value directly?

BTW it’s a BC break to use it at third argument, it needs to be the fourth argument in the constructor

Reply:
Just using codePeriod would allow large discrepancies on each side of the codePeriod.

For example:
codePeriod = 300;
getCode() returns a hash valid for 300 second periods since Unix epoch. Using the discrepancies that exist currently in checkCode(), that means for a code generated at say 10:10:00 would still be valid at 10:19:59 since the discrepancy would allow for the next period as well (periods being 10:10:00, 10:15:00, 10:20:00, etc... as defined by the codePeriod). The bufferPeriod defines the period size (I could rename it to this if helpful) so that the discrepancies would only allow a code generated at 10:10:00 to be valid until 10:15:29.

Reply:
Great explanation, thank you, can you please make the phpdoc more meaningful with explanation of what the parameters are doing?